daemon/logger/amqp/amqp.go

```diff
@@ -0,0 +1,419 @@
+		logrus.Errorf("Could not reconnect: %v", err)
+		return err
+	}
+	logrus.Info("Reconnected")
+	s.connection = connection
+	return nil
+}
+
+// Take the log message and publish it to the currently connected broker
+func (s *amqpLogger) Log(msg *logger.Message) (err error) {
+	// Remove trailing and leading whitespace
+	short := bytes.TrimSpace([]byte(msg.Line))
+
+	if s.connection == nil {
+		err = reconnect(s)
+		if err != nil {
+			return err
+		}
+	}
+
+	currentBroker := s.connection.brokerURLs[s.connection.broker]
+
+	// If the message isn't empty then send it to the broker
+	if string(short) != "" {
+		m := amqpMessage{
+			Version:   "1",
+			Host:      s.fields.Hostname,
+			Message:   string(short),
+			Timestamp: time.Now(),
+			Path:      s.fields.ContainerID,
+			Tags:      s.fields,
+		}
+
+		messagejson, err := json.Marshal(m)
+		if err != nil {
+			logrus.Errorf("Could not serialise event - %v", err)
+			return err
+		}
+
+		amqpmsg := amqp.Publishing{
+			DeliveryMode: amqp.Persistent,
+			Timestamp:    time.Now(),
+			ContentType:  "application/json",
+			Body:         messagejson,
+		}
+
+		if currentBroker.Confirm == true && s.connection != nil {
+			defer confirmOne(s.connection.conf)
+		}
+
+		err = s.connection.c.Publish(currentBroker.Exchange, currentBroker.RoutingKey, false, false, amqpmsg)
+		if err != nil {
+			err = reconnect(s)
+			if err != nil {
+				return err
+			}
+		}
+
+	}
+
+	return nil
+}
+
+// If message confirmation is enabled then ensure that the confirmation message is
+// received and dealt with
+func confirmOne(confirms <-chan amqp.Confirmation) {
+	logrus.Debug("Waiting for confirmation of publishing")
+	if confirmed := <-confirms; confirmed.Ack {
+		logrus.Debugf("Confirmed delivery with tag: %d", confirmed.DeliveryTag)
+	} else {
+		logrus.Debugf("Failed delivery with tag: %d", confirmed.DeliveryTag)
+	}
+}
+
+// Take the log-driver options and put them into a data structure. Convert any values
+// to the appropriate data type (eg string -> bool)
+func parseURL(ctx logger.Context) (brokerArray []*amqpBroker) {
+	amqpURLs := strings.Split(ctx.Config["amqp-url"], " ")
+	for _, amqpURL := range amqpURLs {
+		u, err := url.Parse(amqpURL)
+		b, err := strconv.ParseBool(ctx.Config["amqp-confirm"])
+		if err != nil {
+			logrus.Errorf("String %v is not a valid URL: %v", u, err)
+		} else {
+			broker := &amqpBroker{
+				BrokerURL:  u,
+				CertPath:   ctx.Config["amqp-cert"],
+				KeyPath:    ctx.Config["amqp-key"],
+				Exchange:   ctx.Config["amqp-exchange"],
+				Queue:      ctx.Config["amqp-queue"],
+				RoutingKey: ctx.Config["amqp-routingkey"],
+				Tag:        ctx.Config["amqp-tag"],
+				Confirm:    b,
+			}
+			brokerArray = append(brokerArray, broker)
+		}
+	}
+	return brokerArray
+}
+
+// Take the settings specified in the JSON configuration file and put them into a data
+// structure. Convert any values to their required data types
+func parseJSONFile(path string) (brokerArray []*amqpBroker) {
```

Comment:
I think file could be pretty confusing for cli users, because it's file on server-side and you're not always have control over server, so it will be impossible to configure this for some people.